src/components/dialogs/commons/criteria-based/criteria-based-utils.js

```diff
@@ -106,26 +135,79 @@ export const CONTINGENCY_LIST_EQUIPMENTS = {
     },
 };
 
-export const getCriteriaBasedSchema = (id = CRITERIA_BASED) => ({
-    [id]: yup.object().shape({
+export const FILTER_EQUIPMENTS = {
+    ...CONTINGENCY_LIST_EQUIPMENTS,
+    THREE_WINDINGS_TRANSFORMER: {
+        id: 'THREE_WINDINGS_TRANSFORMER',
+        label: 'ThreeWindingsTransformers',
+        fields: [countries, nominalVoltage1, nominalVoltage2, nominalVoltage3],
+    },
+    GENERATOR: {
+        id: 'GENERATOR',
+        label: 'Generators',
+        fields: [countries, energySource, nominalVoltage],
+    },
+    LOAD: {
+        id: 'LOAD',
+        label: 'Loads',
+        fields: [countries, nominalVoltage],
+    },
+    BATTERY: {
+        id: 'BATTERY',
+        label: 'Batteries',
+        fields: [countries, nominalVoltage],
+    },
+    LCC_CONVERTER_STATION: {
+        id: 'LCC_CONVERTER_STATION',
+        label: 'LccConverterStations',
+        fields: [countries, nominalVoltage],
+    },
+    VSC_CONVERTER_STATION: {
+        id: 'VSC_CONVERTER_STATION',
+        label: 'VscConverterStations',
+        fields: [countries, nominalVoltage],
+    },
+    VOLTAGE_LEVEL: {
+        id: 'VOLTAGE_LEVEL',
+        label: 'VoltageLevels',
+        fields: [countries, nominalVoltage],
+    },
+    SUBSTATION: {
+        id: 'SUBSTATION',
+        label: 'Substations',
+        fields: [countries, nominalVoltage],
+    },
+};
+
+export const getCriteriaBasedSchema = (extraFields) => ({
+    [CRITERIA_BASED]: yup.object().shape({
+        [COUNTRIES]: yup.array().of(yup.string()),
         [COUNTRIES_1]: yup.array().of(yup.string()),
         [COUNTRIES_2]: yup.array().of(yup.string()),
+        ...getRangeInputSchema(NOMINAL_VOLTAGE),
         ...getRangeInputSchema(NOMINAL_VOLTAGE_1),
         ...getRangeInputSchema(NOMINAL_VOLTAGE_2),
+        ...extraFields,
     }),
 });
 
-export const getCriteriaBasedFormData = (
-    id = CRITERIA_BASED,
-    countries1 = [],
-    countries2 = [],
-    nominalVoltage1 = DEFAULT_RANGE_VALUE,
-    nominalVoltage2 = DEFAULT_RANGE_VALUE
-) => ({
-    [id]: {
-        [COUNTRIES_1]: countries1,
-        [COUNTRIES_2]: countries2,
-        ...getRangeInputDataForm(NOMINAL_VOLTAGE_1, nominalVoltage1),
-        ...getRangeInputDataForm(NOMINAL_VOLTAGE_2, nominalVoltage2),
+export const getCriteriaBasedFormData = (criteriaValues, extraFields) => ({
+    [CRITERIA_BASED]: {
+        [COUNTRIES]: criteriaValues?.[COUNTRIES] ?? [],
+        [COUNTRIES_1]: criteriaValues?.[COUNTRIES_1] ?? [],
+        [COUNTRIES_2]: criteriaValues?.[COUNTRIES_2] ?? [],
+        ...getRangeInputDataForm(
+            NOMINAL_VOLTAGE,
+            criteriaValues?.[NOMINAL_VOLTAGE] ?? DEFAULT_RANGE_VALUE
```

Comment:
```suggestion
            criteriaValues?.[NOMINAL_VOLTAGE]
```
DEFAULT_RANGE_VALUE is already the default value
Same for calls below